client/asset/btc/rpcclient.go

```diff
@@ -256,6 +256,20 @@ func (wc *rpcClient) reconfigure(cfg *asset.WalletConfig, currentAddress string)
 			return false, errors.New("wrong net")
 		}
 
+		// Ensure we can get a new address and decode it.
+		if wc.addrFunc != nil { // with an addrFunc, cannot check since it likely uses the prior rpcClient
```

Comment:
The trailing comment here is actually a minor pitfall with the new approach of having the `addrFunc` implementation as a closure that captures the parent `asset.Wallet` concrete type, although I still like it in general.